skbeam/core/correlation.py

```diff
@@ -1113,77 +1167,6 @@ def _cross_corr(img1, img2=None):
     # fftconvolve(A,B) = FFT^(-1)(FFT(A)*FFT(B))
     # but need FFT^(-1)(FFT(A(x))*conj(FFT(B(x)))) = FFT^(-1)(A(x)*B(-x))
     reverse_index = [slice(None, None, -1) for i in range(ndim)]
-    imgc = fftconvolve(img1, img2[reverse_index], mode='same')
+    imgc = fftconvolve(img1, img2[reverse_index], mode='full')
```

Comment:
It's true, but it works out here. I should explain this. So previously, the way I ran the computation using (`ifft(fft(a)*conj(fft(b)) )`), I had to double the image size, then run the convolution mentioned. The reason for doubling is to ensure that boundary points are not correlated together. 

When I switched to `fftconvol` as you suggested previously, I expanded the image and used `mode='same'` to ensure the output was same size as this doubled array.

However, what i did not appreciate (which Mark Sutton pointed out) is that `fftconvol` does this magic for you. It doubles your image (plus a few extra pixels to make it a magic number) to prevent this wraparound. Thus, expanding the image and using `method='same'` was more or less doing the same thing twice. All is needed to do is not expand the image, let fftconvol do it for you and set `method='full'`.

So in light of that, after dropping image doubling, I had to select `method=full` to ensure the image is roughly the same size. However, in this case, `fftconvol` is smarter than me and the array is actually consistently slightly smaller (i.e. it wraps the boundaries as tight as possible).

I had done testing and saw the array sizes were off by one or two pixels (but never more). I plotted both the 1D and 2D cases and compared to previous data and they're almost the same.

I hope I explained correctly? It is easier to explain in person.